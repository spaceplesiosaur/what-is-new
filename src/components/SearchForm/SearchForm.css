.navBar {
  position: absolute;
  vertical-align: top;
  text-align: right;
  padding: 2em;
}
