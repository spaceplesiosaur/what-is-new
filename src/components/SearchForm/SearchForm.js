import React from 'react';
import './SearchForm.css';

class SearchForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      searchInput: ""
    }
  }
  updateInput = event => {
    this.setState({searchInput: event.target.value})
  }

  filterStories = (props) => {
    this.props.filterArticles(this.state.searchInput);
  }

  render() {
    return (
      <section className="navBar">
        <input type="text" className="searchBox" onChange={event => this.updateInput(event)}/>
        <button onClick={this.filterStories} tabIndex="0">Search Articles</button>
      </section>
    )
  }
}

export default SearchForm;
