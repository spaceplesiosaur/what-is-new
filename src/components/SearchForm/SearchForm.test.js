import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should match the snapshot with correct data', () => {
    const wrapper = shallow(<SearchForm
    filterArticles={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
