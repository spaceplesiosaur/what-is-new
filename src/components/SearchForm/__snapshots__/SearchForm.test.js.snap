// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with correct data 1`] = `
<section
  className="navBar"
>
  <input
    className="searchBox"
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</section>
`;
